[MU3] fix #296528: Footers now appear inside page margins

SKefalidis opened the pull request:
Resolves: https://musescore.org/en/node/296528

Added the ability to layout a textbase using the bottom margin (the old way used the top margin in every case). This change fixes the issue.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [n\a] I created the test (mtest, vtest, script test) to verify the changes I made

dmitrio95 commented:
As mentioned also in the issue, this seems to effectively reverse the change made in #4944, although the implementation seems to be different.

SKefalidis commented:
Updated and rebased. Added the option of how to draw the footers as a style.

![image](https://user-images.githubusercontent.com/21060365/109849776-706ca000-7c5a-11eb-9be2-121b0292370e.png)

Jojo-Schmitz commented:
Adds one string to the translations. Not an issue, just noting...

RobFog commented:
@dmitrio95 What do you think makes more sense?

dmitrio95 commented:
Well, I am not that sure what makes more sense. The change made by this PR makes both footers and headers lie outside of the margins area which is indeed more consistent than the situation we have now. In this regard I think this change makes sense.

As this PR introduces a new style setting, `footerInsideMargins`, it may also make sense for compatibility reasons to set it to `true` when importing 2.X and perhaps older 3.X scores (this seems to be determined now by the files in [`share/styles`](https://github.com/musescore/MuseScore/blob/e37696b1102bb9dbb3c93c2be874c5e5cc9ce3cd/share/styles/legacy-style-defaults-v2.mss)). Otherwise this change, while fixing the inconsistent default settings for newer scores, also effectively returns [this issue](https://musescore.org/en/node/285361) back.

Jojo-Schmitz commented:
I can't see any effect when toggling that option "Inside margins", it still stays outside the page margins?

Seems 2.x has it outside the margins, (2.3.2 at least). 3.5.2 too (and I have no older 3.x handy to test), so no need to tweak the style files.

Jojo-Schmitz commented:
I take it all back, I just accidentally left out one line of you PR, and of course (Murphy's Law) the vital one from textbase.cpp...
It does seem to place it way too high though?

Still no style default change needed for any prior version as far as I can tell. Or would this be needed only needed for 3.0?

RomanPudashkin commented:
3.x is closed for any changes
